src/main/java/com/myhome/controllers/PaymentController.java

```diff
@@ -0,0 +1,184 @@
+    PaymentDto scheduledPayment = paymentService.schedulePayment(requestPaymentDto);
+
+    if (scheduledPayment == null) {
+      return  ResponseEntity.status(HttpStatus.NOT_FOUND).body(new SchedulePaymentResponse());
+    }
+
+    SchedulePaymentResponse scheduledPaymentResponse =
+        schedulePaymentApiMapper.paymentToSchedulePaymentResponse(scheduledPayment);
+
+    return ResponseEntity.status(HttpStatus.CREATED).body(scheduledPaymentResponse);
+  }
+
+  @Operation(description = "Get details about a payment with the given payment id")
+  @GetMapping(
+      path = "/members/payments/{paymentId}",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<SchedulePaymentResponse> listPaymentDetails(
+     @PathVariable String paymentId) {
+    Optional<PaymentDto> paymentOptional = paymentService.getPaymentDetails(paymentId);
+    log.trace("Received request to get details about a payment with id[{}]", paymentId);
+
+    if (!paymentOptional.isPresent()) {
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new SchedulePaymentResponse());
+    }
+    PaymentDto paymentDto = paymentOptional.get();
+
+    SchedulePaymentResponse schedulePaymentResponse =
+        schedulePaymentApiMapper.paymentToSchedulePaymentResponse(paymentDto);
+    return ResponseEntity.status(HttpStatus.OK).body(schedulePaymentResponse);
+  }
+
+  //need to implement functionality to get all payments details for a specified member
+  @Operation(description = "Get all payments for the specified member")
+  @GetMapping(
+    path = "/members/{memberId}/payments",
+    produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<ListMemberPaymentsResponse> listAllMemberPayments(
+      @PathVariable String memberId) {
+    log.trace("Received request to list all the payments for the house member with id[{}]", memberId);
+
+    HouseMember member = paymentService.getHouseMember(memberId);
+
+    if (member == null) {
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new ListMemberPaymentsResponse());
+    }
+
+    Set<Payment> memberPayments =
+        member.getPayments();
+
+    Set<ListMemberPaymentsResponse.MemberPayment> getMemberPaymentResponseSet =
+        schedulePaymentApiMapper.memberPaymentSetToRestApiResponseMemberPaymentSet(memberPayments);
+
+    ListMemberPaymentsResponse response = new ListMemberPaymentsResponse();
+    response.getPayments().addAll(getMemberPaymentResponseSet);
+    return ResponseEntity.status(HttpStatus.OK).body(response);
+  }
+
+  @Operation(description = "Get all payments scheduled by the specified admin")
+  @GetMapping(
+    path = "/communities/{communityId}/admins/{adminId}/payments",
+    produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<ListMemberPaymentsResponse> listALlAdminScheduledPayments(
```

Comment:
There is a typo in method name `ALl` -> `All`

Reply:
Ya I caught that